src/plugins/ui_core_views/cell_evaluation/evaluation_plugin.ts

```diff
@@ -269,6 +271,14 @@ export class EvaluationPlugin extends UIPlugin {
     );
   }
 
+  getArrayFormulaSpreadingOn(position: CellPosition): CellPosition | undefined {
```

Comment:
getCorrespondingFormulaCell can do the trick (it will be available after a small rebase)